apps/nova/src/components/dashboard/PastLessons.tsx

```diff
@@ -53,8 +83,8 @@ export const PastLessons = () => {
   });
 
   const lessons = useMemo(
-    () => asLessons(lessonsQuery.list),
-    [lessonsQuery.list]
+    () => asLessons(lessonsQuery.list, user?.role),
+    [lessonsQuery.list, user?.role]
   );
```

Comment:
No need to make user role optional. If no user role, render nothing. 

```suggestion
  const lessons = useMemo(
    () => user.role ? asLessons(lessonsQuery.list, user.role) : [],
    [lessonsQuery.list, user?.role]
  );
```